Snowplow/GlobalContext/SPSchemaRule.m

```diff
@@ -0,0 +1,153 @@
+//
+//  SNOWSchemaRule.m
+//  Snowplow-iOS
+//
+//  Copyright (c) 2013-2020 Snowplow Analytics Ltd. All rights reserved.
+//
+//  This program is licensed to you under the Apache License Version 2.0,
+//  and you may not use this file except in compliance with the Apache License
+//  Version 2.0. You may obtain a copy of the Apache License Version 2.0 at
+//  http://www.apache.org/licenses/LICENSE-2.0.
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the Apache License Version 2.0 is distributed on
+//  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+//  express or implied. See the Apache License Version 2.0 for the specific
+//  language governing permissions and limitations there under.
+//
+//  Authors: Alex Benini
+//  Copyright © 2020 Snowplow Analytics. All rights reserved.
+//  License: Apache License Version 2.0
+//
+
+#import "SPSchemaRule.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SPSchemaRule ()
+
+@property (nonatomic, copy, readwrite) NSString *rule;
+@property (nonatomic, copy, readwrite) NSArray<NSString *> *ruleParts;
+
+@end
+
+@implementation SPSchemaRule
+
+static NSString * const kRulePattern = @"^iglu:((?:(?:[a-zA-Z0-9-_]+|\\*)\\.)+(?:[a-zA-Z0-9-_]+|\\*))\\/([a-zA-Z0-9-_\\.]+|\\*)\\/([a-zA-Z0-9-_\\.]+|\\*)\\/([1-9][0-9]*|\\*)-(0|[1-9][0-9]*|\\*)-(0|[1-9][0-9]*|\\*)$";
```

Comment:
Sometimes I come across a Regex pattern and I have to wonder if it's the best solution to the problem. Don't get me wrong, I love Regex but when it ends up like this, it's pretty much unreadable although as a solution it certainly solves the rule matching succinctly. 
I'm not saying this needs to change by the way, I just wanted to express an opinion :-)

Reply:
I agree the readability of regex is awful. I've copied this from the previous implementation and I preferred to keep that there as it just works. In case of a further complication it makes sense to split it or reconsider the logic.